### PR TITLE
Split Vulnerability and Save platforms

### PR DESCRIPTION
**Pull request by Cheshiriks**
Снимок экрана 2023-11-27 в 13 12 02

Split Vulnerability and Save platforms

### CODE REVIEW

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/topbar/TopBar.kt
Comment (akuleshov7) [inline save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/topbar/TopBar.kt:31]: What I am thinking now is that we should have different TopBars, I guess.

**Review by akuleshov7 — APPROVED** [review]
After TopBar split we can merge it, but @Cheshiriks please monitor and  block all PRs and SAVE updates. Because the next step is the separate deployment

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/App.kt
Comment (akuleshov7) [inline save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/App.kt:7]: Do not forget to update COSV views before merging (there were changes from @DrAlexD on the frontend side of COSV). And looks like you have forgotten them

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/CosvSchemaView.kt
Comment (akuleshov7) [inline save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/CosvSchemaView.kt:5]: delete vuln directory from the SAVE frontend, that will prevent the confusion and invalid changes

File: .run/Cosv-Frontend-LocalRun.run.xml
Comment (akuleshov7) [inline .run/Cosv-Frontend-LocalRun.run.xml:1]: where do you put resources (static) like images and gifs for COSV?

Comment (akuleshov7) [reply .run/Cosv-Frontend-LocalRun.run.xml:1]: they should also be split

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/topbar/TopBar.kt
Comment (Cheshiriks) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/topbar/TopBar.kt]: Splited it

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/App.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/App.kt:10]: Updated

**Comment by kgevorkyan** [comment]
could add github action for building the `save-cosv-frontend` ?

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/CosvSchemaView.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/CosvSchemaView.kt:5]: Deleted

**Comment by Cheshiriks** [comment]
> could you add github action for building the `save-cosv-frontend` ?

Added

File: .run/Cosv-Frontend-LocalRun.run.xml
Comment (Cheshiriks) [reply .run/Cosv-Frontend-LocalRun.run.xml:1]: Removed unnecessary images from _cosv-frontend_

File: save-cosv-frontend/README.md
Comment (kgevorkyan) [inline save-cosv-frontend/README.md:1]: does this file duplicates the readme from `save-frontend`?

Comment (kgevorkyan) [inline save-cosv-frontend/README.md:1]: does this file duplicates the readme from `save-frontend`?

Comment (Cheshiriks) [reply save-cosv-frontend/README.md:1]: Yes. The frontend for COSV contains the same instructions as the Save frontend. The only difference is in the launch command.

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/fileuploader/CosvFileManagerComponent.kt
Comment (nulls) [inline save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/fileuploader/CosvFileManagerComponent.kt:1]: why it's moved to the common module?

Comment (akuleshov7) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/fileuploader/CosvFileManagerComponent.kt:1]: @Cheshiriks , please speed up and fix these problems

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/UserBoard.kt
Comment (akuleshov7) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/UserBoard.kt:33]: I am afraid it will be difficult for him now, this PR is still not finished - my suggestion to do that, but in a separate PR asap as soon as he will merge this one and we can unlock other people who are blocked by Vlad now

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/VulnerabilityInfoTab.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/VulnerabilityInfoTab.kt:319]: No, that's right. I don't know why GitHub displays it this way. But it was always there: `affectedCarousel`

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/views/vuln/UploadCosvButton.kt
Comment (Cheshiriks) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/views/vuln/UploadCosvButton.kt:1]: Because this button is located in the header of the table with a list of vulnerabilities. And we use the table of vulnerabilities both in the view with Users and in the view with Organizations.

File: save-cosv-frontend/README.md
Comment (kgevorkyan) [reply save-cosv-frontend/README.md:1]: may be leave one readme in `save-frontend-common`, and link readmes from `save-frontend` and `save-cosv-frontend` to this common readme? like see 'save-frontend-common/readme'?

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/views/vuln/UploadCosvButton.kt
Comment (Cheshiriks) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/views/vuln/UploadCosvButton.kt:1]: We will make different views for organizations in one of the following projects, then I will move these files

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/RequestModalHandler.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/RequestModalHandler.kt:129]: Fixed

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/topbar/TopBarLinks.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/topbar/TopBarLinks.kt:29]: Removed

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/CreateVulnerabilityView.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/CreateVulnerabilityView.kt:467]: Removed

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/VulnerabilityHeader.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/VulnerabilityHeader.kt:147]: Removed

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt:335]: Removed

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/welcome/pagers/WelcomePager.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/views/welcome/pagers/WelcomePager.kt:11]: Moved it

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/routing/MobileRouting.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/routing/MobileRouting.kt:20]: Deleted

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/TimelineComponent.kt
Comment (Cheshiriks) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/TimelineComponent.kt:17]: Deleted

File: save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/fileuploader/CosvFileManagerComponent.kt
Comment (Cheshiriks) [reply save-frontend-common/src/main/kotlin/com/saveourtool/save/frontend/common/components/basic/fileuploader/CosvFileManagerComponent.kt:1]: Moved it to cosv-frontend

File: save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/RequestModalHandler.kt
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/kotlin/com/saveourtool/save/cosv/frontend/components/RequestModalHandler.kt:26]: Yes, that's why I had to copy it, it uses different TopBars

File: save-cosv-frontend/README.md
Comment (Cheshiriks) [reply save-cosv-frontend/README.md:1]: Done. I will delete duplicate information from Readme in the `save-frontend` in the next PR.

**Review by kgevorkyan — APPROVED** [review]

File: save-cosv-frontend/src/main/resources/img/sad_cat.png
Comment (Cheshiriks) [reply save-cosv-frontend/src/main/resources/img/sad_cat.png:1]: I'll move them in the next PR.

**Pull request merged**
